### PR TITLE
ci: setup GitHub actions

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

First step of #4507

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

First step to make sure everything is set up properly :)
This will run CircleCI and GitHub Actions side by side.

Some stats:

- Circle CI affect build (PR) 12m 28s - https://app.circleci.com/pipelines/github/ngrx/platform/3713/workflows/55e21343-4028-4ae5-8df5-34c3b6ab4ce0
- GitHub affected build (PR) 16m 18s - https://github.com/ngrx/platform/actions/runs/10742124526
- GitHub affected build (PR) 7m 43s - https://github.com/ngrx/platform/actions/runs/10743257281

- CircleCI full build (main) 12m 28s - https://app.circleci.com/pipelines/github/ngrx/platform/3696/workflows/8cce591c-5094-4f1b-97c9-67444c675823
- GitHub full build (main) 18m 0s - https://github.com/ngrx/platform/actions/runs/10742124522

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+name: main
```
Comment (timdeschryver) [inline .github/workflows/main.yml:1]: I enabled this to test it out, here's a successful run:

https://github.com/ngrx/platform/actions/runs/10742124522

File: projects/ngrx.io/src/karma.conf.js [hunk 2]
```diff
@@ -25,14 +25,8 @@ module.exports = function (config) {
     colors: true,
     logLevel: config.LOG_INFO,
     autoWatch: true,
-    browsers: ['CustomChrome'],
+    browsers: ['ChromeHeadless'],
```
Comment (timdeschryver) [inline projects/ngrx.io/src/karma.conf.js:28]: Required, otherwise the CI needs to install chromium (which we did in CircleCI via an orb)

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks Tim!

**Review by rainerhahnekamp — COMMENTED** [review]
Looks very structured, very clean. 👍  Really nice!

The configuration for Circle CI will be removed in this PR or in another one?

**Comment by timdeschryver** [comment]
> The configuration for Circle CI will be removed in this PR or in another one?

Yea, it's my intention to keep them in parallel for a while. After v19 we can fully migrate and include the release step.

**Review by rainerhahnekamp — APPROVED** [review]
As said before, well structured and if something doesn't work out, we still have CircleCI as fallback. 👍

**Pull request merged**